lib/Entities/NewsletterEntity.php

```diff
@@ -131,25 +131,25 @@ class NewsletterEntity {
 
   /**
    * @ORM\OneToMany(targetEntity="MailPoet\Entities\NewsletterEntity", mappedBy="parent")
-   * @var NewsletterEntity[]|ArrayCollection
+   * @var ArrayCollection<int, NewsletterEntity>
```

Comment:
I'm not sure about removing `NewsletterEntity[]`. 

`NewsletterEntity[]` is technically incorrect annotation but it is [a trick](https://stackoverflow.com/a/14725444) how to make code completion work in PHPStorm.

Without it we loose completion like this one:
<img width="976" alt="Screen Shot on 2021-07-29 at 13-34-51" src="https://user-images.githubusercontent.com/1082140/127487086-d993b742-334c-46b1-9696-099a0669ad54.png">

On the other hand the completion doesn't work when accessing Collection elements via Collection methods e.g.
`$this->children->get(0)->get...` even with the trick.

I tried adding back `NewsletterEntity[]` into an annotation and it seems that PHP Stan doesn't mind it so we could still allow the PHP Stan rule and also keep the trick.

I'm inclined to keep the trick there so that it is comfortable for developers to work with the collections. Doctrine collections [can be used as arrays](https://github.com/doctrine/common/blob/2.1.x/lib/Doctrine/Common/Collections/Collection.php#L46) so if we use them like that we could still have the advantage of code completion.

It is not a strong opinion though so I guess we could also ask others about their thoughts. What do you think @pavel-mailpoet , @lysyjan , @brezocordero ?

Reply:
I agree with you @costasovo

Reply:
Good point, @costasovo. I wasn't sure why we had `NewsletterEntity[]` in the docblock as I believe this is not the correct type for this property. Thanks for clarifying the reason.

When I first read your comment, I was torn. On the one hand, I understand the importance of auto-completion in the IDE, but on the other hand, I wasn't sure if we should use the incorrect type because our IDE doesn't support a specific PHPDoc feature.

I dig a little deeper and found that support for generics is coming in the next version of PHPStorm (https://blog.jetbrains.com/phpstorm/2021/07/phpstorm-2021-2-beta/), which I imagine will be released soon. The RC is already out: https://www.jetbrains.com/phpstorm/nextversion/. I believe this means that once the new version is released, auto-completion will work with just `@var ArrayCollection<int, NewsletterEntity>`.

In my opinion, this changes things a little bit, and I'm more inclined to suggest that we keep `@var ArrayCollection<int, NewsletterEntity>` instead of adding `NewsletterEntity[]` back. What do you and the others think?

Reply:
@rodrigoprimo That's great news! I din't know it. In that case I'm for going with correct annotations and no hacks.